### PR TITLE
Revert to using CLU15 message for Gear Selection for Compatability Reasons

### PR DESCRIPTION
**Pull request by emmertex**
Issues with using LVR12 message are as follows.

On 8 Speed, if in drive, but switched to "sport" mode, it no longer see's drive
On vehicles equipped with paddles for gear selections, pressing a paddle is seen as leaving drive.
Hyundai Eleantra does not create this message.

None of these issues are present when using the CLU15 message, while still allowing the same featureset.

### CODE REVIEW

**Comment by jfrux** [comment]
Umm okay... but…. Why?
"Sport" mode really isn't for the openpilot world.  Is it really called “Sport” mode?… or are you talking about “Sequential” which is like a paddle-shifted mode?

I’m not sure that it makes sense for OP to allow usage in Sequential mode on any car.  Sequential mode isn’t to be used ever unless well… you think you’re somehow benefiting from shifting the car manually… of which you probably picked the wrong car to begin with and probably think self-driving cars are silly because IMA CONTROL MAH OWN CAR CUZZ I’M MAN AND MAN CONTROL CAR AND MACHINE CAN SUCK IT.

Then again... what do I know? I'll tell you... nothing.  Carry on.
:) <3

**Comment by jfrux** [comment]
IMHO… that has nothing to do with Openpilot.
Just my 2 cents...

**Comment by emmertex** [comment]
It's an ACC issue.
Because ACC will ride the brakes for almost an entire descent, you will have no brakes at the bottom. Everyone dies.
If you force 1 gear downshift, ACC will ride the brakes 1/2 as much, you will have brakes at the bottom. 

There is no other reason OP can't fully drive those roads, but a forced downshift will force a manual drive.

The auto is happy to sit at 3000rpm for ascent, but won't auto down shift to allow more than 2250rpm on descent.  The downshift is not sufficient on these vehicles.  
 
OP should not be restricted due to bad auto design.

**Comment by rbiasini** [comment]
@emmertex , As we discussed, can you please keep the Santa Fe the way it was? Ideally, I prefer listening to the gear from the engine module rather than the CLU, so we should find that source on all Kia/hyundai

File: selfdrive/car/hyundai/interface.py [hunk 1]
```diff
@@ -186,7 +186,10 @@ def update(self, c):
     ret.wheelSpeeds.rr = self.CS.v_wheel_rr
 
     # gear shifter
-    ret.gearShifter = self.CS.gear_shifter
+    if candidate == CAR.ELANTRA:
```
Comment (rbiasini) [inline selfdrive/car/hyundai/interface.py]: self.candidate?

**Comment by rbiasini** [comment]
if somebody can test this on a Elantra and a non-Elantra, I'll merge it.
Thanks!

**Comment by emmertex** [comment]
Merged changes made by TK211X
Tested on Sorento as well for non-elantra test

**Comment by rbiasini** [comment]
@emmertex, if you give the green light (tested on Elantra too) I would like to merge this.

**Comment by emmertex** [comment]
Yes, all tested and good to go.
Was tested by me and @TK211X

**Pull request merged**